### PR TITLE
Add PNG support

### LINKED ISSUE #19: Add PNG support

### PR DESCRIPTION
**Pull request by sanks64**
Closes #19

Docs:

- https://github.com/randy408/libspng/blob/master/examples/example.c
- https://libspng.org/

### CODE REVIEW

**Comment by sanks64** [comment]
|                            | Time     | Memory  |
| -------------------------- | -------- | ------- |
| bilinear_resize!(640, 480) | 6.13ms   | 1.17MiB |
|           brightness!(1.4) | 1.79ms   | 0B      |
|              box_blur!(10) | 7.83ms   | 600kiB  |
| channel_swap!(:red, :blue) | 74.0ns   | 0B      |
|             contrast!(128) | 2.05ms   | 0B      |
|         gaussian_blur!(10) | 23.12ms  | 1.75MiB |
|       horizontal_blur!(10) | 3.32ms   | 300kiB  |
|         vertical_blur!(10) | 4.36ms   | 300kiB  |
|  crop!(200, 200, 100, 100) | 48.96µs  | 102kiB  |
|    to_jpeg(IO::Memory.new) | 3.8ms    | 964kiB  |
|     to_png(IO::Memory.new) | 37.32ms  | 1.3MiB  |
|     to_ppm(IO::Memory.new) | 2.39ms   | 2.0MiB  |
|    to_webp(IO::Memory.new) | 106.39ms | 1.23MiB |

Total Time: 198.54ms
Total Memory: 9.6MiB

File: src/pluto/bindings/lib_spng.cr [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  fun ctx_new = spng_ctx_new(flags : CtxFlags) : Void*
+  fun decode_image = spng_decode_image(ctx : Void*, out : Void*, len : LibC::SizeT, fmt : Format, flags : DecodeFlags) : LibC::Int
+  fun decoded_image_size = spng_decoded_image_size(ctx : Void*, fmt : Format, len : LibC::SizeT*) : LibC::Int
+  fun encode_chunks = spng_encode_chunks(ctx : Void*) : LibC::Int
+  fun encode_image = spng_encode_image(ctx : Void*, img : Void*, len : LibC::SizeT, fmt : Format, flags : EncodeFlags) : LibC::Int
+  fun get_ihdr = spng_get_ihdr(ctx : Void*, ihdr : IHDR*) : LibC::Int
+  fun get_png_buffer = spng_get_png_buffer(ctx : Void*, len : LibC::SizeT*, error : LibC::Int*) : UInt8*
+  fun set_ihdr = spng_set_ihdr(ctx : Void*, ihdr : IHDR*) : LibC::Int
+  fun set_option = spng_set_option(ctx : Void*, option : Option, value : LibC::Int) : LibC::Int
+  fun set_png_buffer = spng_set_png_buffer(ctx : Void*, buf : Void*, size : LibC::SizeT) : LibC::Int
```
Comment (Vici37) [inline src/pluto/bindings/lib_spng.cr:75]: `lib_jpeg_turbo` has its `fun`s at the top of the file followed by the enums. Do we want to keep that pattern for other lib files? Minor question on styling for consistency sake, it doesn't matter to me one way or the other.

File: src/pluto/format/png.cr [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    image_data = IO::Memory.new(size * 4)
+    size.times do |index|
+      image_data.write_byte(red.unsafe_fetch(index))
+      image_data.write_byte(green.unsafe_fetch(index))
+      image_data.write_byte(blue.unsafe_fetch(index))
+      image_data.write_byte(alpha.unsafe_fetch(index))
+    end
+
+    ctx = LibSPNG.ctx_new(LibSPNG::CtxFlags::Encoder)
+    raise Exception.new("Failed to create a context") unless ctx
```
Comment (Vici37) [inline src/pluto/format/png.cr:63]: We have both `::Pluto::Exception` and `Exception` in this class - should we consolidate to one pattern? At first I read this as a vanilla exception and was confused on why you weren't just `raise "Failed to create a context"` :P

File: src/pluto/exception.cr [hunk 3]
```diff
@@ -1,5 +1,9 @@
 class Pluto::Exception < ::Exception
-  getter error_code : Int32
+  getter error_code : Int32?
+
+  def initialize(message : String | Nil = nil, cause : Exception | Nil = nil)
+    super(message, cause)
```
Comment (Vici37) [inline src/pluto/exception.cr:5]: We probably want to ensure at least `message` or `cause` are non-nil for the sake of providing useful information to the developer on why the exception was thrown.

**Review by Vici37 — APPROVED** [review]
Minor styling questions, nothing blocking or problematic. Great job! PNG support will be good. I see WebP is still "winning" the perf charts...

**Pull request merged**